### ISSUE TITLE
Possible Virus in Companion App

### ISSUE THREAD

**Issue by AdvantBeiten** [QuestPDF/QuestPDF#1005]
I can't install the latest version of the companion app, because it identifies as being infected with a virus. Could you look into this? Thanks!

**Comment by sbsrouteur** [comment]
Same here, trellix report malware when running the 2014.10.4 installer.
Did not have the issue with previous versions.

**Comment by MarcinZiabek** [comment]
Windows and Linux versions are built directly from the source code on GitHub actions. There are very few code changes between 2024.10.3 and 2024.10.4. I suspect it could be a false positive. 

I am also preparing to publish the source code of the Companion App, but it may take a week or two...

I am happy to collaborate with you on this issue.

**Comment by girlpunk** [comment]
Are you able to share the reported signatures that these virus scanners identify the companion app as?

**Comment by sbsrouteur** [comment]
Signature is : FP Artemis!D4ADAA77FCC9

**Comment by sbsrouteur** [comment]
HI,

I did report a false positive to trellix, and could install companion 2024.10.4.

**Comment by w4l1k** [comment]
![Questpng](https://github.com/user-attachments/assets/db639a36-8135-41e4-bc7e-3f56e2604254)
Today's installation

**Comment by IanWitham** [comment]
I've been running the latest version fine, and then today it suddenly won't work. "Operation did not complete successfully because the file contains a virus or potentially unwanted software".  The exe and shortcuts are immediately deleted when this happens. Even the installer exe is being periodically deleted for some reason.

I did install Windows updates yesterday, so that may be the cause.

Adding the folder and the exe to my Windows Defender exclusions doesn't work. Very frustrating.

**Comment by AdvantBeiten** [comment]
I immediatly get a virus warning when downling the new version 10.6. This is really frustrating.
![Virus Quest Pdf](https://github.com/user-attachments/assets/8da588d6-1313-40b1-909b-774ce60d57c1)

**Comment by MarcinZiabek** [comment]
The source code of QuestPDF Companion App has been published on GitHub and is publicly available here: https://github.com/QuestPDF/QuestPDF.Companion

All Windows and Linux builds are done via GitHub Actions, which can be reviewed here: https://github.com/QuestPDF/QuestPDF.Companion/actions/workflows/build.yml

MacOS build is done locally via Xcode.

I hope that this action will ensure you that the build is trustworthy. I am very open to any feedback on mitigating the antivirus problem. Thank you!

**Comment by sbsrouteur** [comment]
You can report a false positive to trellix. They update their signature relatively quickly. This is what I did after reporting the issue with version 2024.10.4

**Comment by v3nus-py** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by htuomas** [comment]
I've been using the Companion app succesfully for a week or so, and suddenly today Windows Defender abruptly closed it and deleted the program. I tried downloading and installing again, but to no avail. 

"Defender detected and quarantined active 'Trojan:Script/Wacatac.H!ml' in file 'questpdf_companion.exe', during attempted open by 'svchost.exe'"

**Comment by sbsrouteur** [comment]
> I've been using the Companion app succesfully for a week or so, and suddenly today Windows Defender abruptly closed it and deleted the program. I tried downloading and installing again, but to no avail.
> 
> "Defender detected and quarantined active 'Trojan:Script/Wacatac.H!ml' in file 'questpdf_companion.exe', during attempted open by 'svchost.exe'"

Heuristic reputation rules in AV suck. to work around this, I : 
  - Disable AV
  - Download, install App
  - Whitelist App exe & folder in AV
  - Restart AV

Repeat as many time as needed :(

**Comment by blogcraft** [comment]
Adding a Digital Sign to the installer could also be a good practice.

**Comment by MarcinZiabek** [comment]
I completely agree! I’m planning to purchase a Code Signing EV certificate, as I believe it will enhance the developer experience and reduce the risk of antivirus issues.

That said, I’m a bit surprised at how challenging these solutions are to integrate with CI/CD workflows. The options seem to be either USB tokens that require physical machines, or expensive cloud-based solutions.

That being said, the process is already underway!

**Comment by blogcraft** [comment]
Also, It could be nice if the QuestPDF Companion could be distributed by the Microsoft store. That way is an easy to download and also the store managing auto updating would be nice.

**Comment by MarcinZiabek** [comment]
There has been some delay in this effort, but I’m happy to share that it’s back on the roadmap! 🎉

I’m currently investigating which code-signing certificate needs to be purchased to increase trust in the QuestPDF Companion App and reduce the likelihood of false-positive antivirus scan results.

Since there has been no recent activity, I’m closing this issue. If you have a similar request in the future, feel free to open a new GitHub issue. Thanks for your feedback! ❤️
